numpy/ma/core.py

```diff
@@ -5869,6 +5869,18 @@ def __getitem__(self, indx):
 
         """
         m = self._mask
+        if isinstance(m[indx], ndarray):
```

Comment:
I think this line is good now.

But for any future reference, another possibility which you reminded me of in another PR is `if m.dtype[indx].subdtype is not None`, which checks specifically for subarrays. But I think a check for `ndarray` is better because there might be other ways of getting an ndarray when indexing - for instance, multifield indices like `marr[['f0', 'f1']]` could return an ndarray, which we would want to convert to a masked array. (Currently void scalars to not support multifield indices though, but I think that is an oversight which we might want to fix soon, eg in #6053)